python/shotgun_desktop/upgrade_startup.py

```diff
@@ -106,9 +84,20 @@ def upgrade_startup(splash, sgtk, app_bootstrap):
     out_of_date = (latest_descriptor.get_version() != current_desc.get_version())
 
     if not out_of_date:
-        logger.debug("Desktop startup does not need upgrading. Currenty running version %s" % current_desc.get_version())
+        logger.debug(
+            "Desktop startup does not need upgrading. Currenty running version %s" % current_desc.get_version()
+        )
+        return False
+
+    # FIXME: Violating internal API
+    from sgtk.commands.console_utils import _check_constraints
```

Comment:
Again, this is bad, but it invoked the bundle core, so it doesn't introduce something we need to keep supporting forever. Maybe that check should be exposed on the descriptor itself @manneohrstrom ?